### DISCUSSION TITLE
404 error during last step of Provider login process to get a JWT

### DISCUSSION THREAD

**Discussion by nitewulf** [strapi/strapi#7321] [Legacy Topics]
I'm looking at the pending doc as updated here:

https://github.com/strapi/strapi/pull/7132/files?file-filters%5B%5D=.js&file-filters%5B%5D=.md

and see that my problem has been that I needed my frontend app to exchange the Providers JWT for a Strapi JWT.

7. The frontend (`http://website.com/connect/github/redirect`) calls the backend with `https://strapi.website.com/auth/github/callback?access_token=eyfvg` that returns the strapi user profile with its `jwt`. <br> (Under the hood, the backend asks Github for the user's profile and a match is done on Github user's email address and Strapi user's email address)

I'm getting a 404 error from my frontend application trying to POST to mystrapiwebsite.com/auth/google/callback.  I checked the routes.json file and don't see any handler for a /auth/:provider/callback

is that logic part of a pending PR?

**Comment by derrickmehaffy** [comment]
Hi @nitewulf that referenced route comes from the users-permissions plugin here: https://github.com/strapi/strapi/blob/a8bb45da52a0f7e8e3a47c07f306a3662d8e0cc1/packages/strapi-plugin-users-permissions/config/routes.json#L233-L246

Which calls this controller: https://github.com/strapi/strapi/blob/a8bb45da52a0f7e8e3a47c07f306a3662d8e0cc1/packages/strapi-plugin-users-permissions/controllers/Auth.js#L22-L177

Can you give us some more information about your environment such as the following?

```
**System**

- Node.js version: <!-- Please ensure you are using the Node LTS version (v12) -->
- NPM version:
- Strapi version: 
- Database:
- Operating system:
```

**Reply by nitewulf** [reply]
Node.js - 12.1
NPM - my DBA is looking
Strapi - 3.0.6
Database - postgress 11
OS - Docker running on Ubuntu 18.0.4

I do see that GET route.  But this step, from looking at the updated documentation, looks like it should be a POST as it is sending the Providers issued JWT access token.  Or am I still not understanding this process?

**Reply by nitewulf** [reply]
I did update my front end app to send the JWT with a GET request and got back a strapi JWT and user information.  Have to say, I'm super pumped!!!   Just concerned about the security of that.  Shouldn't the Providers JWT be passed back to strapi via POST?

**Reply by petersg83** [reply]
Hi :) If you use https, the GET parameters are encrypted so no worry on this side. However, it's true that the provider `access_token` may appear in the server logs, which is not ideal. A POST request might be better. What do you think @alexandrebodin ?

**Reply by derrickmehaffy** [reply]
I certainly agree @petersg83 a POST request would be better
